pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -546,12 +546,22 @@ func (ctrl *Controller) syncContainerRuntimeConfig(key string) error {
 			}
 			if isNotFound {
 				tempIgnCfg := ctrlcommon.NewIgnConfig()
-				mc = mtmpl.MachineConfigFromIgnConfig(role, managedKey, &tempIgnCfg)
+				rawTempIgnCfg, err := json.Marshal(tempIgnCfg)
+				if err != nil {
+					glog.V(2).Infoln(tempIgnCfg, err, "error marshalling new Ignition config: %v", err)
+				}
+				mc = mtmpl.MachineConfigFromRawIgnConfig(role, managedKey, rawTempIgnCfg)
 			}
-			mc.Spec.Config = createNewIgnition(map[string][]byte{
+
+			internalCtrRuntimeConfigIgn := createNewIgnition(map[string][]byte{
 				storageConfigPath: storageTOML,
 				crioConfigPath:    crioTOML,
 			})
+			rawCtrRuntimeConfigIgn, err := json.Marshal(internalCtrRuntimeConfigIgn)
+			if err != nil {
+				glog.V(2).Infoln(internalCtrRuntimeConfigIgn, err, "error marshalling container runtime config Ignition: %v", err)
+			}
+			mc.Spec.Config.Raw = rawCtrRuntimeConfigIgn
```

Comment:
The “update mc.Spec.Config” operation might also be better with a helper (although saving one line and one variable, it’s less clear cut).